### DISCUSSION TITLE
How to cast uuid to text or varchar with supabase select()

### DISCUSSION THREAD

**Discussion by manholok** [supabase/supabase#3061] [Questions] [answered]
I'm trying to use the ilike() filter on a primary key that is a uuid

**This works:**
```
supabase()
        .from('users')
        .select('*')
        .or(`name.ilike.%${searchTerm}%,email.ilike.%${searchTerm}%`);
```

**This  doesn't work:**
When I try to use user_id, instead of email.

```
supabase()
        .from('users')
        .select('*')
        .or(`name.ilike.%${searchTerm}%,
                user_id.ilike.%${searchTerm}%`);
```

The error I get is:

```
hint: "No operator matches the given name and argument types. You might need to add explicit type casts."
message: "operator does not exist: uuid -> unknown
```

What does it mean to explicit type cast, I do find some SQL examples, so then I would need to dive into creating rpc, but before that, I wanted to know if it is possible with the current select and filters?

**Comment by dshukertjr** [comment, marked as answer]
Hi @manholok!

It seems like this is not possible directly at the moment as[ this issue](https://github.com/PostgREST/postgrest/issues/1409) from PostgREST explains. 

You can achieve what you are trying to do by creating a computed column where you can type cast your `user_id` as text and apply the `ilike` filter on that column!
https://postgrest.org/en/v6.0/api.html#computed-columns
